lib/gceimgutils/gceimgutils.py

```diff
@@ -44,13 +50,38 @@ def __init__(
         except AttributeError:
             self.log_level = self.log.logger.level  # LoggerAdapter
 
-        try:
-            self.credentials = utils.get_credentials(project, credentials_path)
-        except Exception as err:
-            self.log.error(format(err))
+    # ---------------------------------------------------------------------
+    @property
+    def compute_driver(self):
+        """Get an authenticated compute driver"""
+        if not self._compute_driver:
+            self._compute_driver = utils.get_compute_api(self.credentials)
+
+            try:
+                # Force auth to catch errors in one place
+                self._compute_driver.images().list(
+                    project=self.project,
+                    maxResults=1
+                ).execute()
```

Comment:
I think this is too expensive for what we are trying to achieve. The credentials have a `valid` method [1] that shold get us the results we are after. It appears to me that in https://google-auth.readthedocs.io/en/stable/reference/google.oauth2._service_account_async.html#google.oauth2._service_account_async.Credentials.valid we should call `valid` and if that fails raise the exception there.

I am making the assumption that `valid` is available in the version of googleoauth2 we are using. This has to be verified.


[1] https://google-auth.readthedocs.io/en/stable/reference/google.oauth2._service_account_async.html#google.oauth2._service_account_async.Credentials.valid

Reply:
@rjschwei was able to get it down to:

```
credentials = credentials.with_scopes(['profile'])
authed_session = AuthorizedSession(credentials)
response = authed_session.get('https://www.googleapis.com/oauth2/v3/userinfo')
```

This forces token refresh and only returns info about the service account if successful. Otherwise it raises the RefreshError.